src/Compilers/CSharp/Portable/Symbols/SymbolWithAnnotations.cs

```diff
@@ -364,32 +378,34 @@ protected virtual TypeSymbolWithAnnotations SubstituteType(AbstractTypeMap typeM
             var newTypeWithModifiers = typeMap.SubstituteType(this.TypeSymbol, withTupleUnification);
             bool newIsAnnotated = this.IsAnnotated || newTypeWithModifiers.IsAnnotated;
```

Comment:
I'm having trouble thinking of a scenario where `this.IsAnnotated` and `newTypeWithModifiers.IsAnnotated` are different. Is it possible?

Reply:
For type argument substitution in `C<object?>`, `this` is the type parameter `T!` and `newTypeWithModifiers` is `object?`.

---
In reply to: [203895472](https://github.com/dotnet/roslyn/pull/28687#discussion_r203895472) [](ancestors = 203895472)